src/hooks/useTasks.ts

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { ITasksProps } from 'types/tasks';
+import { todoListService } from 'services/api';
+
+interface useTasksProps {
+  tasks: ITasksProps[];
+  addTask: (title: string, description: string, isCompleted: boolean) => void;
+  completeTask: (taskId: string) => void;
+  deleteTask: (taskId: string) => void;
+}
+
+const useTasks = (): useTasksProps => {
+  const [tasks, setTasks] = useState<ITasksProps[]>([]);
+
+  useEffect(() => {
+    const loadTasks = async () => {
+      try {
+        const tasksData = await todoListService();
+        setTasks(tasksData);
+      } catch (error) {
+        console.error('Erro ao carregar lista de tarefas:', error);
+      }
+    };
+
+    loadTasks();
+  }, []);
+
+  const addTask = (title: string, description: string, isCompleted: boolean) => {
+    if (title.trim() && description.trim() !== '') {
+      const newTask = {
+        id: uuidv4(),
+        title: title,
+        description: description,
+        isCompleted: isCompleted,
+      };
+
+      setTasks([...tasks, newTask]);
```

Comment:
para fazer merge do estado antigo (old state) com o novo, é melhor fazer dessa forma aqui pra evitar bugs:
```suggestion
      setTasks(oldState => [...oldState, newTask]);
```

essa dica é importante, muito pleno e alguns seniors não fazem isso direito, e aprender isso de uma vez vai ser muito útil